### ISSUE TITLE
Error: Encrypted private key detected, but no passphrase given

### ISSUE THREAD

**Issue by eighteight** [green-arrow/ember-cli-deploy-ssh-index#1]
I am using the config from the example:

```
module.exports = function(deployTarget) {

  var ENV = {
    build: {
      environment: deployTarget
    },
    'revision-data': {
      type: 'git-commit'
    },
    'ssh-index': {
      remoteDir: "/app/myapp",
      username: "user",
      host: "host",
      privateKeyFile: "/Users/user/.ssh/id_rsa",
      allowOverwrite: true
    },
    rsync: {
      dest: "/app/myapp",
      username: "user",
      host: "host",
      delete: false
    }
  };

  return ENV;

};
```

```
...
+- upload
|  |
|  +- rsync
|    - Uploading using rsync...
|    - 
|  |
|  +- ssh-index
|    - Attempting to connect to remote host: user@host:22
|
+- didFail
Error: Encrypted private key detected, but no passphrase given
Error: Encrypted private key detected, but no passphrase given
    at Client.connect (/Users/user/myapp/node_modules/ember-cli-deploy-ssh-index/node_modules/ssh2/lib/client.js:151:15)
    at /Users/user/myapp/node_modules/ember-cli-deploy-ssh-index/lib/ssh.js:151:14
    at lib$rsvp$$internal$$initializePromise (/Users/user/myapp/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:1084:9)
    at PromiseExt.lib$rsvp$promise$$Promise (/Users/user/myapp/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:546:53)
    at new PromiseExt (/Users/user/myapp/node_modules/ember-cli/lib/ext/promise.js:32:8)
    at CoreObject.module.exports.CoreObject.extend._connect (/Users/user/myapp/node_modules/ember-cli-deploy-ssh-index/lib/ssh.js:137:12)
    at CoreObject.module.exports.CoreObject.extend._establishSftp (/Users/user/myapp/node_modules/ember-cli-deploy-ssh-index/lib/ssh.js:107:17)
    at CoreObject.module.exports.CoreObject.extend.fetchRevisions (/Users/user/myapp/node_modules/ember-cli-deploy-ssh-index/lib/ssh.js:81:17)
    at CoreObject.module.exports.CoreObject.extend.upload (/Users/user/myapp/node_modules/ember-cli-deploy-ssh-index/lib/ssh.js:32:17)
    at Class.DeployPluginBase.extend.upload (/Users/user/myapp/node_modules/ember-cli-deploy-ssh-index/index.js:56:20)|
Pipeline aborted
```

**Comment by green-arrow** [comment]
@eighteight - It looks like I forgot to document it, but you should be able to provide your private key's passphrase by providing a `passphrase` configuration property:

```
'ssh-index': {
  remoteDir: "/app/myapp",
  username: "user",
  host: "host",
  privateKeyFile: "/Users/user/.ssh/id_rsa",
  passphrase: "YourPassphrase", // or better, process.ENV.YOUR_PASSPHRASE_VAR
  allowOverwrite: true
}
```

Let me know if that solves your problem. I will make sure to update the Readme to point out that option.

**Comment by eighteight** [comment]
@green-arrow I tried that but got the same error. What fixed it for me is rebuilding the ssh keys and leaving the passphrase empty.

Thanks!

**Comment by green-arrow** [comment]
@eighteight - Glad you got it working!

I am going to leave this issue open and try and look into verifying that passphrase protected key files work as expected.

**Comment by mazondo** [comment]
Had the same issue.  This fixed it for me: https://github.com/green-arrow/ember-cli-deploy-ssh-index/pull/2

**Comment by green-arrow** [comment]
Fixed by #2

**Comment by green-arrow** [comment]
`passphrase` support (along with `agent` support) is now available in [v0.2.1](https://github.com/green-arrow/ember-cli-deploy-ssh-index/releases/tag/v0.2.1)
